railgun/manager/flagset.go

```diff
@@ -1,17 +1,82 @@
 package manager
 
 import (
-	"github.com/kong/kubernetes-ingress-controller/pkg/util"
+	"fmt"
+
 	"github.com/spf13/pflag"
+
+	"github.com/kong/kubernetes-ingress-controller/pkg/annotations"
+	"github.com/kong/kubernetes-ingress-controller/pkg/util"
 )
 
-// flagSet extends flag.FlagSet with additional variable types.
-type flagSet struct {
-	pflag.FlagSet
-}
+// -----------------------------------------------------------------------------
+// Controller Manager - CLI Flagset Configuration
+// -----------------------------------------------------------------------------
 
 // EnablementStatusVar defines a flag of type EnablementStatus.
 func (f *flagSet) EnablementStatusVar(p *util.EnablementStatus, name string, value util.EnablementStatus, usage string) {
 	*p = value
 	f.Var(p, name, usage)
 }
+
```

Comment:
My intention was for `flagset.go` to be for the `type flagSet` definition.

I would put `MakeFlagSetFor` in `config.go`, because `flagset.go` is about the definition of the flagset machinery (defining a wrapper over `pflag.FlagSet` with custom flag types), and `config.go` is about the knobs that configure the mamager.

Suggestion: move `MakeFlagSetFor` to `config.go`. Maybe even make it a method of `type Config`?

Reply:
sounds good :+1: 1bc713e96791d9a0ed9b2a633a20e038e53fb5b0 resolves